tests/accumulator_test.py

```diff
@@ -118,31 +121,71 @@ def test_serialization_with_incompatible_serialized_object(self):
 class GenericAccumulatorTest(unittest.TestCase):
 
     def test_merge_accumulators(self):
-        mean_accumulator1 = MeanAccumulator().add_value(15)
-        mean_accumulator2 = MeanAccumulator().add_value(5)
+        mean_accumulator1 = MeanAccumulator(params=[], values=[15])
+        mean_accumulator2 = MeanAccumulator(params=[], values=[5])
 
         merged_accumulator = accumulator.merge(
             [mean_accumulator1, mean_accumulator2])
 
         self.assertEqual(merged_accumulator.compute_metrics(), 10)
 
     def test_merge_diff_type_throws_type_error(self):
-        mean_accumulator1 = MeanAccumulator().add_value(15)
-        sum_squares_acc = SumOfSquaresAccumulator().add_value(1)
+        mean_accumulator1 = MeanAccumulator(params=[], values=[15])
+        sum_squares_acc = SumOfSquaresAccumulator(params=[], values=[1])
 
         with self.assertRaises(TypeError) as context:
             accumulator.merge([mean_accumulator1, sum_squares_acc])
+
         self.assertIn("The accumulator to be added is not of the same type."
                       "", str(context.exception))
 
+    @patch('pipeline_dp.accumulator.create_accumulator_params')
+    def test_accumulator_factory(self, mock_create_accumulator_params_function):
+        aggregate_params = pipeline_dp.AggregateParams([agg.Metrics.MEAN], 5, 3)
+        budget_accountant = pipeline_dp.BudgetAccountant(1, 0.01)
+
+        values = [[10]]
+        mock_create_accumulator_params_function.return_value = [
+            accumulator.AccumulatorParams(MeanAccumulator, None)
+        ]
```

Comment:
could you please add check that `mock_create_accumulator_params_function` is called with correct parameters. That might be done for example with [assert_called_with](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with)

Reply:
Done!